Transportation/Bike/BikeHireDockingStation/harvest/bicycle_hire_station_barcelona_harvest.py

```diff
@@ -0,0 +1,169 @@
+        print('Error while calling: %s : %s' % (source, e))
+        if f is not None:
+            f.close()
+        return None
+
+
+# Harmonise the station data
+def harmonize_station(station_data):
+    out = {
+        'type': BIKE_STATION_TYPE,
+        'id': 'Bcn-BikeHireDockingStation-' + station_data['id'],
+        'freeSlotNumber': {
+            'type': 'Number',
+            'value': int(station_data['slots'])
+        },
+        'availableBikeNumber': {
+            'type': 'Number',
+            'value': int(station_data['bikes'])
+        },
+        'address': {
+            'type': 'PostalAddress',
+            'value': {
+                'addressCountry': 'ES',
+                'addressLocality': 'Barcelona',
+                'streetAddress': sanitize(station_data['streetName'] +
+                                          ',' + station_data['streetNumber'])
+            }
+        },
+        'location': {
+            'type': 'geo:json',
+            'value': {
+                'type': 'Point',
+                'coordinates': [
+                    float(station_data['longitude']),
+                    float(station_data['latitude']),
+                    float(station_data['altitude'])
+                ]
+            }
+        },
+        'status': {
+            'type': 'Text',
+            'value': status_dictionary[station_data['status']]
+        }
+    }
+
+    current_timestamp = datetime.now(barcelona_tz)
+
+    out['freeSlotNumber']['metadata'] = {
+        'timestamp': {
+            'type': 'DateTime',
+            'value': current_timestamp.replace(microsecond=0).isoformat()
+        }
+    }
+    out['availableBikeNumber']['metadata'] = out['freeSlotNumber']['metadata']
+
+    return out
+
+
+# Persists the data to a Data Broker supporting FIWARE NGSI v2
+def persist_data(entity_list):
+    # print json.dumps(entity_list)
+
+    data_obj = {
+        'actionType': 'APPEND',
+        'entities': entity_list
+    }
+    data_as_str = json.dumps(data_obj)
+
+    headers = {
+        'Content-Type': MIME_JSON,
+        'Content-Length': len(data_as_str),
+        'Fiware-Service': FIWARE_SERVICE,
+        'Fiware-Servicepath': FIWARE_SERVICE_PATH
+    }
+
+    req = urllib2.Request(
+        url=(
+            DATA_BROKER +
+            '/v2/op/update'),
+        data=data_as_str,
+        headers=headers)
+    f = None
+    try:
+        f = urllib2.urlopen(req)
+        f.close()
+        print('Entities successfully created')
+    except urllib2.URLError as e:
+        print('Error while POSTing data to Orion: %d %s' % (e.code, e.read()))
+
+
+# Main module
+def main():
+    data = read_data()
+
+    if data is None:
+        exit()
+
+    parsed_data = json.loads(data)
+
+    station_list = parsed_data['stations']
+
+    ngsi_data = []
+
+    for station in station_list:
+        h_station = harmonize_station(station)
+        ngsi_data.append(h_station)
+
+    persist_data(ngsi_data)
+
```

Comment:
```python
def main():
    data = read_data()
    if data:
        persist_data(harmonize_station(station) for station in json.loads(data)['stations'])
    else:
        print("read_data() failed!')
```

Reply:
too long line -1

Reply:
sorry but I still think code is clearer as it is now